src/NUnitFramework/framework/Internal/Commands/TestMethodCommand.cs

```diff
@@ -47,7 +47,10 @@ public override TestResult Execute(TestExecutionContext context)
             if (_testMethod.HasExpectedResult)
                 Assert.That(result, Is.EqualTo(_testMethod.ExpectedResult));
 
-            context.CurrentResult.SetResult(ResultState.Success);
+            if (context.MultipleAssertLevel > 0)
```

Comment:
question: Should we dispose of any open scopes here after reporting the error? I'm not sure if there could be any stability issues with leaving them around given they themselves contain a reference to a particular `TestExecutionContext`

Reply:
We cannot as we don't have a reference to them. I don't think it is worth it to add them as references to the _context_.
Every test gets it [own new context](https://github.com/nunit/nunit/blob/main/src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs#L272) and the old scope and context will be garbage collected.

Reply:
Ok sounds good. My thinking had indeed been probing if it made sense to track these on the context as a `Stack<AssertionScope>` instead of the `MultipleAssertLevel` counter, but as you say we can also fulfill the behaviour without doing those broader changes.